core/src/main/java/org/apache/iceberg/MetadataUpdate.java

```diff
@@ -490,4 +491,21 @@ public void applyTo(ViewMetadata.Builder viewMetadataBuilder) {
       viewMetadataBuilder.setCurrentVersionId(versionId);
     }
   }
+
+  class AppendFilesUpdate implements MetadataUpdate {
```

Comment:
We might want to rename this to `AppendManifestsUpdate` as appending files would be more of a `DataFile` type object.

Reply:
Actually, I think it makes more sense to pass DataFile entries through this interface since that will actually handle both cases.  There are issues with passing manifests because they will need to be rewritten in certain cases.  [ContentFileParser](https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/ContentFileParser.java#L55) already supports a json representation, so I think we should pass the data file records through this interface directly (as opposed to just pointing to files).

Reply:
> I think it makes more sense to pass DataFile entries through this interface since that will actually handle both cases
+1. I think using manifest was more for scalability considerations because there might be many data files. But that might be an over-optimization, and start with passing data file entries is a better starting point.

Reply:
For the naming, should this be more like `AddContentFileUpdate` instead of `AppendContentFileUpdate` in this case? @danielcweeks given this can be used to add both data and delete files which are both content files, and "append delete files" does not sound like the right terminology.

Reply:
I think we need to initially disallow delete files from being added because there are more concerns around achieving correct semantics.  We could split this into `AppendDataFileUpdate` and if we later decide to support deletes, add `AppendDeleteFileUpdate`.